packages/plugins/plugin-facebook-app-events/src/FacebookAppEventsPlugin.tsx

```diff
@@ -0,0 +1,35 @@
+import {
+  DestinationPlugin,
+  PluginType,
+  TrackEventType,
+} from '@segment/analytics-react-native';
+import track from './methods/track';
+import type { SegmentClient } from '@segment/analytics-react-native/src/analytics';
+
+import { Settings } from 'react-native-fbsdk-next';
+
+export class FacebookAppEventsPlugin extends DestinationPlugin {
+  type = PluginType.destination;
+  key = 'FacebookAppEvents';
+
+  async configure(analytics: SegmentClient) {
+    this.analytics = analytics;
+    let adTrackingEnabled =
+      this.analytics?.store.getState().main.context?.device?.adTrackingEnabled;
```

Comment:
yeah we also need a better way to update plugins if/when it becomes available because I'm not sure this is actually working correctly (will get to it when your comments are all resolved

Reply:
I wonder if there is a way to incorporate consent manager and idfa plugins easily into dest plugins

Reply:
like as an option